rolling-shutter/gnosisaccessnode/decryptionkeyshandler.go

```diff
@@ -39,25 +44,49 @@ func (handler *DecryptionKeysHandler) ValidateMessage(_ context.Context, msg p2p
 	return pubsub.ValidationAccept, nil
 }
 
-func (handler *DecryptionKeysHandler) validateCommonFields(key *p2pmsg.DecryptionKeys) (pubsub.ValidationResult, error) {
-	if key.InstanceID != handler.config.InstanceID {
+func (handler *DecryptionKeysHandler) validateCommonFields(keys *p2pmsg.DecryptionKeys) (pubsub.ValidationResult, error) {
+	if keys.InstanceID != handler.config.InstanceID {
 		return pubsub.ValidationReject,
-			errors.Errorf("instance ID mismatch (want=%d, have=%d)", handler.config.InstanceID, key.GetInstanceID())
+			errors.Errorf("instance ID mismatch (want=%d, have=%d)", handler.config.InstanceID, keys.GetInstanceID())
 	}
-	if key.Eon > math.MaxInt64 {
-		return pubsub.ValidationReject, errors.Errorf("eon %d overflows int64", key.Eon)
+	if keys.Eon > math.MaxInt64 {
+		return pubsub.ValidationReject, errors.Errorf("eon %d overflows int64", keys.Eon)
 	}
 
-	if len(key.Keys) == 0 {
+	if len(keys.Keys) == 0 {
 		return pubsub.ValidationReject, errors.New("no keys in message")
 	}
-	if len(key.Keys) > int(handler.config.MaxNumKeysPerMessage) {
+	if len(keys.Keys) > int(handler.config.MaxNumKeysPerMessage) {
 		return pubsub.ValidationReject, errors.Errorf(
 			"too many keys in message (%d > %d)",
-			len(key.Keys),
+			len(keys.Keys),
 			handler.config.MaxNumKeysPerMessage,
 		)
 	}
+
+	eonKey, ok := handler.storage.GetEonKey(keys.Eon)
+	if !ok {
+		return pubsub.ValidationReject, errors.Errorf("no eon key found for eon %d", keys.Eon)
+	}
+
+	for i, k := range keys.Keys {
```

Comment:
We should not forget to also apply optimizations here at one point, or ideally remove code redundancies.